src/apps/chifra/pkg/pinlib/manifest/tab_manifest.go

```diff
@@ -0,0 +1,77 @@
+package manifest
+
+import (
+	"encoding/csv"
+	"io"
+)
+
+func ReadPinDescriptors(r io.Reader) ([]PinDescriptor, error) {
+	reader := csv.NewReader(r)
+	reader.Comma = '\t'
+	reader.FieldsPerRecord = 3
+
+	// source, err := reader.ReadAll()
+	descriptors := []PinDescriptor{}
+
+	for {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		descriptors = append(descriptors, PinDescriptor{
+			FileName:  record[0],
+			BloomHash: record[1],
+			IndexHash: record[2],
+		})
+	}
+
+	return descriptors, nil
+}
+
+func BuildTabRange(descriptors []PinDescriptor) (ManifestRange, error) {
+	firstPinRange, err := ManifestRangeFromString(descriptors[0].FileName)
```

Comment:
Hm, this would be an empty file, right? Maybe we should have a validator for manifest file...
Change this function to check `len` and return error if it's 0